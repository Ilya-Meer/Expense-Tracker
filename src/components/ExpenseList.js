import React from 'react';
import { connect } from 'react-redux';
import { StyledExpenseList } from './StyledComponents/HomeDashboard';
import getVisibleExpenses from '../selectors/expenses';
import ExpenseListItem from './ExpenseListItem';

export const ExpenseList = (props) => {
  
  return (
    <StyledExpenseList>
      <h2>Expense List</h2>
      <div>
        {props.expenses.length === 0 ? (
          <p>Add an Expense!</p>
         ) : (
          props.expenses.map(expense => {
            return  <ExpenseListItem key={expense.id} {...expense} />
          }))
        } 
      </div>
    </StyledExpenseList>
  )
}


const mapStateToProps = (state) => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);