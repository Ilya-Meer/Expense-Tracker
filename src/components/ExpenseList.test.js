import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from './ExpenseList';
import moment from 'moment';

describe('Expense List', () => {
  const expenses = [{
    id: '3',
    desc: 'Rent', 
    note: 'The rent was not cheap either', 
    amount: '2000000', 
    createdAt: moment(0)
  }];


  it('should render expense list with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
  })

  it('should render expense list with prompt if given no expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
  })
})