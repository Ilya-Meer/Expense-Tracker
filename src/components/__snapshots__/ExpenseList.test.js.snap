// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense List should render expense list with expenses 1`] = `
<styled.div>
  <h2>
    Expense List
  </h2>
  <div>
    <ExpenseListItem
      amount="2000000"
      createdAt={"1970-01-01T00:00:00.000Z"}
      desc="Rent"
      id="3"
      key="3"
      note="The rent was not cheap either"
    />
  </div>
</styled.div>
`;

exports[`Expense List should render expense list with prompt if given no expenses 1`] = `
<styled.div>
  <h2>
    Expense List
  </h2>
  <div>
    <p>
      Add an Expense!
    </p>
  </div>
</styled.div>
`;
